NoSQL/Task1.md

```diff
@@ -0,0 +1,402 @@
+
+(1 rows)
+```
+
+Next we Will explore some Other 
+CQL commands that will come in handy, like TRUNCATE in a later exercise, we will show you how to add/remove (non-primary key) columns. 
+## 11. Let's remove the data from our table using TRUNCATE .truncate videos; 
+
+```
+cassandra@cqlsh:killrvideo> TRUNCATE TABLE videos;
+```
+## 12. Create a second table in the 'killrvideo' keyspace called videos_by_title_year with the structure shown in above table. 
+Be sure users can query this table on both 'title' and 'added_year' by combining them into the partition key. 
+
+```
+cassandra@cqlsh:killrvideo> CREATE TABLE IF NOT EXISTS 
+videos_by_title_year (
+    title text,
+    added_year int,
+    added_date timestamp,
+    description text,
+    user_id uuid,
+    video_id timeuuid,
+    PRIMARY KEY ((title, added_year))
+    );
+```
+
+## 13. Load the data from the file using the 'COPY' command 
+
+
+```
+cassandra@cqlsh:killrvideo> COPY videos_by_title_year  FROM '/usr/NoSQL_files/exercise-3/videos_by_title_year.csv' WITH HEADER = true ;
+Detected 4 core(s)
+Using 3 child processes
+
+Starting copy of killrvideo.videos_by_title_year with columns [title, added_year, added_date, description, user_id, video_id].
+Closing queues...ws; Rate:     830 rows/s; Avg. rate:     830 rows/s
+Closing queues...
+Closing queues...
+Processed: 430 rows; Rate:     415 rows/s; Avg. rate:     695 rows/s
+430 rows imported from 1 files in 0.619 seconds (0 skipped).
+```
+
+## 15. Try running queries on the table to query on a specific 'title' and 'added_year'. 
+
+Query by both added_year and title:
+
+```
+cassandra@cqlsh:killrvideo> SELECT * FROM videos_by_title_year WHERE title = 'ASP.NET Community Standup - April 7th, 2015' and added_year = 2015 ;
+
+ title                                       | added_year | added_date                      | description | user_id                              | video_id
+---------------------------------------------+------------+---------------------------------+-------------+--------------------------------------+--------------------------------------
+ ASP.NET Community Standup - April 7th, 2015 |       2015 | 2015-04-09 08:00:02.000000+0000 |        null | a284468d-d20c-4745-8e20-5125386f2025 | 872f8113-22e9-1f31-9dfd-05ae3222feb7
+
+(1 rows)
+```
+
+Try to query by added_year:
+
+```
+cassandra@cqlsh:killrvideo> SELECT * FROM videos_by_title_year WHERE title = 'ASP.NET Community Standup - April 7th, 2015' and year = 2015 ;
```

Comment:
The point here was to query just part of the composite PK. But here you show wrong query, "year" column is not exists.